python/cac_tripplanner/destinations/models.py

```diff
@@ -171,3 +218,69 @@ class ExtraDestinationPicture(ExtraImage):
 
 class ExtraEventPicture(ExtraImage):
     event = models.ForeignKey('Event')
+
+
+class UserFlagSummaryManager(models.Manager):
+    """Annotate queryset of attractions to add user flag count summaries."""
+    def get_queryset(self):
+        queryset = super(UserFlagSummaryManager, self).get_queryset()
+        for flag, label in UserFlag.UserFlags.CHOICES:
+            queryset = queryset.annotate(**{flag: Coalesce(Subquery(
+                UserFlag.objects.filter(historic=False, is_event=False, flag=flag,
```

Comment:
`is_event=False` here doesn't seem right - shouldn't that be conditional on whether we're looking at events or destinations?

Reply:
Good catch, I'd intended to move this out to the subclasses and vary it there.